### DISCUSSION TITLE
Add custom delay bars to start doing beatmap searches

### DISCUSSION THREAD

**Discussion by huyenden** [ppy/osu#23653] [Ideas] [answered]
For those with a huge number of ranked beatmaps (my 25k+), the problem in both the stable and lazer versions is that when I search for a beatmap, when I haven't entered all the search keywords, the beatmap search speed of osu to render too quickly. This leads to osu having to scan 2-3 times for the same keyword. This causes more time consuming

For example, when I want to find the beatmap MariannE, osu will in turn show the search results for Ma... Mari... Mariann... MariannE. Each scan will scan through 25k+ beatmaps with a delay of 3s each time, it will take about 10s to complete the search.

If there is enough time needed to enter keywords about 1.5 seconds before starting the search will be better. :3
![image](https://github.com/ppy/osu/assets/128424762/0f9313b2-ea2d-475f-a93d-db10f8c20536)

**Comment by peppy** [comment, marked as answer]
Eventually we'll move the search processing to the background, so this will no longer be an issue.
